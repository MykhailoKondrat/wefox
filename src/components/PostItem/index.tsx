import {
  Box,
  Button,
  Card,
  CardActions,
  CardContent,
  CardMedia,
  Typography,
} from "@mui/material";
import { FC } from "react";
import { ToggleManagePost } from "../../features/ManagePost/hooks/useManagePost";

export interface PostItemProps {
  title: string;
  content: string;
  image_url?: string;
  lat?: string;
  long?: string;
  toggleEditMode: ToggleManagePost;
  deletePost: () => void;
}

export const PostItem: FC<PostItemProps> = ({
  title,
  image_url,
  content,
  toggleEditMode,
  deletePost,
}) => (
  <Box mb={2} pb={1}>
    <Card>
      <CardMedia component="img" alt={title} height="auto" image={image_url} />
      <CardContent>
        <Typography gutterBottom variant="h5" component="div">
          {title}
        </Typography>
        <Typography variant="body2" color="text.secondary">
          {content}
        </Typography>
      </CardContent>
      <CardActions>
        <Button
          size="small"
          onClick={() => {
            toggleEditMode();
          }}
        >
          Edit
        </Button>
        <Button
          size="small"
          onClick={() => {
            deletePost();
          }}
        >
          Delete
        </Button>
      </CardActions>
    </Card>
  </Box>
);

export default PostItem;
