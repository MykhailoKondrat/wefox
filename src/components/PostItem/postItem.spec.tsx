import {fireEvent, render, screen} from '@testing-library/react';
import PostItem from "./index";
import { expect, jest } from "@jest/globals";
import React from "react";

describe("Components / PostItem", () => {
  it("should render correctly", () => {
    const { asFragment } = render(
      <PostItem
        title={"test"}
        content={"test"}
        lat={'123'}
        long={'123'}
        image_url={'image-url'}
        toggleEditMode={() => {}}
        deletePost={() => {}}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("Should trigger edit callback on click to edit button", () => {
    const spyEdit = jest.fn()
    render(
      <PostItem
        title={"test"}
        content={"test"}
        toggleEditMode={spyEdit}
        deletePost={() => {}}
      />
    );
    fireEvent.click(screen.getByText(/edit/i))
    expect(spyEdit).toHaveBeenCalled()
  })
  it("Should trigger  delete callback on click to  delete button", () => {
    const spyDelete = jest.fn()
    render(
      <PostItem
        title={"test"}
        content={"test"}
        toggleEditMode={() => {}}
        deletePost={spyDelete}
      />
    );
    fireEvent.click(screen.getByText(/delete/i))
    expect(spyDelete).toHaveBeenCalled()
  })
});
