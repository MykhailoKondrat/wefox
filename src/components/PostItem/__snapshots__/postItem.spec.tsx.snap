// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / PostItem should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-mngfkd"
    data-cy="post-item"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <img
        alt="test"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
        height="auto"
        src="image-url"
      />
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
        >
          test
        </div>
        <div
          class="css-zfsi42-MuiStack-root"
        >
          <div
            class="css-1gi1ppv-MuiStack-root"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
            >
              Lat: 
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
               123
            </p>
          </div>
          <div
            class="css-1gi1ppv-MuiStack-root"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
            >
              Long: 
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
               123
            </p>
          </div>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          test
        </p>
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
          data-cy="edit-button"
          tabindex="0"
          type="button"
        >
          Edit
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root  css-1ir7th6-MuiButtonBase-root-MuiButton-root"
          data-cy="delete-button"
          tabindex="0"
          type="button"
        >
          Delete
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
